src/CMakeLists.txt

```diff
@@ -283,6 +283,20 @@ rocm_install_targets(
     ${CMAKE_CURRENT_BINARY_DIR}/include
 )
 
+set(COMPILER_NAME ${CMAKE_CXX_COMPILER})
+if(NOT WIN32)
+    if(MIGRAPHX_ENABLE_CPU)
+        # Set the compiler as c++ for CPU compilation in linux
+        set(COMPILER_NAME "c++")
+    else()
+        # For GPU compilation, use amdclang
+        get_filename_component(GET_COMPILER_NAME ${CMAKE_CXX_COMPILER} NAME)
+        if(${GET_COMPILER_NAME} MATCHES "^amdclang")
+            set(COMPILER_NAME ${GET_COMPILER_NAME})
+        endif()
+    endif()# end of CPU/GPU check
+endif()
```

Comment:
There shouldn't be any changes to cmake. The [compiler_src.hpp](https://github.com/ROCm/AMDMIGraphX/blob/develop/src/include/migraphx/compile_src.hpp#L56) should be updated to only use `MIGRAPHX_CXX_COMPILER` macro when its not windows:

```cpp
struct MIGRAPHX_EXPORT src_compiler
{
#ifdef _WIN_32
    fs::path compiler                         = MIGRAPHX_CXX_COMPILER;
#else
    fs::path compiler                         = "c++";
#endif
    std::vector<std::string> flags            = {};
    fs::path output                           = {};
    fs::path launcher                         = {};
    std::string out_ext                       = ".o";
    std::function<fs::path(fs::path)> process = nullptr;
    std::vector<char> compile(const std::vector<src_file>& srcs) const;
};
```

Reply:
Thank you @pfultz2 
Updated the  PR as suggested